clients/src/main/java/org/apache/kafka/clients/admin/internals/ListShareGroupOffsetsHandler.java

```diff
@@ -71,63 +73,122 @@ public AdminApiLookupStrategy<CoordinatorKey> lookupStrategy() {
-                            return data;
+        for (CoordinatorKey coordinatorKey : groupIds) {
+            String groupId = coordinatorKey.idValue;
+            if (response.hasGroupError(groupId)) {
+                handleGroupError(coordinatorKey, response.groupError(groupId), failed, unmapped);
+            } else {
+                Map<TopicPartition, Long> groupOffsetsListing = new HashMap<>();
+                response.data().groups().stream().filter(g -> g.groupId().equals(groupId)).forEach(groupResponse -> {
+                    for (DescribeShareGroupOffsetsResponseData.DescribeShareGroupOffsetsResponseTopic topicResponse : groupResponse.topics()) {
+                        for (DescribeShareGroupOffsetsResponseData.DescribeShareGroupOffsetsResponsePartition partitionResponse : topicResponse.partitions()) {
+                            TopicPartition tp = new TopicPartition(topicResponse.topicName(), partitionResponse.partitionIndex());
+                            if (partitionResponse.errorCode() == Errors.NONE.code()) {
+                                // Negative offset indicates there is no start offset for this partition
+                                if (partitionResponse.startOffset() < 0) {
+                                    groupOffsetsListing.put(tp, null);
+                                } else {
+                                    groupOffsetsListing.put(tp, partitionResponse.startOffset());
+                                }
+                            } else {
+                                log.warn("Skipping return offset for {} due to error {}: {}.", tp, partitionResponse.errorCode(), partitionResponse.errorMessage());
+                            }
                         }
-                    ).collect(Collectors.toList())
-            ).collect(Collectors.toList());
-            completed.put(groupId, data);
+                    }
+                });
+
+                completed.put(coordinatorKey, groupOffsetsListing);
+            }
         }
-        return new ApiResult<>(completed, failed, Collections.emptyList());
+        return new ApiResult<>(completed, failed, unmapped);
     }
 
     private static Set<CoordinatorKey> coordinatorKeys(Collection<String> groupIds) {
         return groupIds.stream()
             .map(CoordinatorKey::byGroupId)
             .collect(Collectors.toSet());
     }
+
+    private void handleGroupError(CoordinatorKey groupId,
+                                  Errors error,
+                                  Map<CoordinatorKey, Throwable> failed,
+                                  List<CoordinatorKey> groupsToUnmap) {
+        switch (error) {
+            case GROUP_AUTHORIZATION_FAILED:
+            case UNKNOWN_MEMBER_ID:
+            case STALE_MEMBER_EPOCH:
+                log.debug("`DescribeShareGroupOffsets` request for group id {} failed due to error {}", groupId.idValue, error);
```

Comment:
Should it be `ListShareGroupOffsets` here and other logs?
```suggestion
                log.debug("`ListShareGroupOffsets` request for group id {} failed due to error {}", groupId.idValue, error);
```

Reply:
This reflects the name of the RPC, and that's `DescribeShareGroupOffsets`.